lib/enkf/enkf_config_node.cpp

```diff
@@ -1039,7 +1039,156 @@ enkf_config_node_type * enkf_config_node_alloc_GEN_PARAM_from_config( const conf
+                                                           const char * min_std_file,
+                                                           const char * init_file_fmt) {
+  enkf_config_node_type * config_node   = NULL;
+  config_node = enkf_config_node_new_gen_kw(node_key, gen_kw_format, forward_init);
+
+
+  enkf_config_node_update_gen_kw(config_node,
+                                 enkf_outfile,
+                                 template_file,
+                                 parameter_file,
+                                 min_std_file,
+                                 init_file_fmt);
+
+  return config_node;
+}
+
+enkf_config_node_type * enkf_config_node_alloc_SURFACE_full(const char * node_key,
+                                                            bool forward_init,
+                                                            const char * output_file,
+                                                            const char * base_surface,
+                                                            const char * min_std_file,
+                                                            const char * init_file_fmt) {
+
+  enkf_config_node_type * config_node = enkf_config_node_new_surface( node_key , forward_init );
+  enkf_config_node_update_surface( config_node , base_surface , init_file_fmt , output_file , min_std_file );
 
+  return config_node;
+}
+
+enkf_config_node_type * enkf_config_node_alloc_FIELD_full(const char * node_key,
+                                                          const char * var_type_string,
+                                                          ecl_grid_type * grid,
+                                                          field_trans_table_type * field_trans_table,
+                                                          const char * ecl_file,
+                                                          const char * enkf_infile,
+                                                          bool forward_init,
+                                                          const char * init_transform,
+                                                          const char * output_transform,
+                                                          const char * input_transform,
+                                                          const char * min_std_file,
+                                                          const char * min_key,
+                                                          const char * max_key,
+                                                          const char * init_file_fmt) {
+
+  int truncation = TRUNCATE_NONE;
+
+  double value_min  = -1;
+  double value_max  = -1;
+
+  if (min_key) {
+      truncation |= TRUNCATE_MIN;
+      value_min   = atof(min_key);
+  }
+
+  if (max_key) {
+      truncation |= TRUNCATE_MAX;
+      value_max   = atof(max_key);
+  }
+  enkf_config_node_type * config_node;
+  if (strcmp(var_type_string , PARAMETER_KEY) == 0) {
```

Comment:
Maybe this logic when selecting param vs general could have been moved out into python?

Reply:
Which would need two C functions, instead of one - will double check it.